### PR TITLE
Refactored configuration structure

### LINKED ISSUE #7: Change repository configuration structure
It would make more sense for the configuration to be:

````
repository:
    repository_1:
        type: doctrine_phpcr_odm
        basepath: /foo/bar
    repository_2:
        type: filesystem
        basedir: /tmp/foo
````

Rather than the somewhat inverted:

````
repository:
    doctrine_phpcr_odm:
        repository_1:
             basepath: /foo/bar
````

The reason it is as it is is because it allows us to have strict configuration. But .. It also limits the extensibility of the bundle so it is probably better to have loose validation in the DI and strict validation at "runtime".

### PR DESCRIPTION
**Pull request by WouterJ**
Fixes #7

### CODE REVIEW

File: DependencyInjection/CmfResourceExtension.php [hunk 1]
```diff
@@ -34,64 +35,120 @@ public function load(array $configs, ContainerBuilder $container)
...
+                throw new InvalidConfigurationException(sprintf(
+                    'Unexpected repository type "%s", known types: %s',
+                    $repository['type'],
+                    implode(', ', array_keys($methods))
+                ));
             }
+
+            $definition = $this->{$methods[$repository['type']]}($repository['options'], $alias);
+            $definition->addTag('cmf_resource.repository', array('type' => $repository['type'], 'alias' => $alias));
+            $container->setDefinition('cmf_resource.repository.' . $repository['type'] . '.' . $alias, $definition);
```
Comment (WouterJ) [inline DependencyInjection/CmfResourceExtension.php]: as repository names are now globally unique, would it make sense to still prefix the service ID with the repository type?

Comment (dantleech) [inline DependencyInjection/CmfResourceExtension.php]: I think it would make sense to remove the type prefix.

**Comment by WouterJ** [comment]
@dantleech I've removed the type from it alias.

Could you please review this PR and merge it when you like the changes? That would enable me to work on #10 and finish the "use resourcebundle for TreeBrowser" PR.

File: DependencyInjection/CmfResourceExtension.php [hunk 2]
```diff
@@ -34,64 +35,120 @@ public function load(array $configs, ContainerBuilder $container)
...
 
-    private function loadRepositories($config, $container)
+    private function loadRepositories(array $repositories, ContainerBuilder $container)
     {
-        foreach (array(
+        $methods = array(
             'composite' => 'createCompositeRepository',
             'doctrine_phpcr' => 'createDoctrinePhpcrRepository',
             'doctrine_phpcr_odm' => 'createDoctrinePhpcrOdmRepository',
             'filesystem' => 'createFilesystemRepository',
```
Comment (dantleech) [inline DependencyInjection/CmfResourceExtension.php]: I think we should ultimately make the DI extension repository-type agnostic, but we can do that later.

File: Tests/Unit/DependencyInjection/fixtures/config.xml [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
-            </cmf_resource:composite-repository>
+        <repository name="articles" type="doctrine_phpcr_odm">
+            <option name="basepath">/cmf/articles</option>
+        </repository>
 
-        </cmf_resource:repository>
-    </cmf_resource:config>
-</container>
+        <repository name="stuff" type="composite">
+            <option name="mounts">
```
Comment (dantleech) [inline Tests/Unit/DependencyInjection/fixtures/config.xml]: Can we not inline this - `<option name="mounts" repository="content" mountpoint="/content" /> ` ?

Comment (WouterJ) [inline Tests/Unit/DependencyInjection/fixtures/config.xml]: if we do that, we can only configure one mount in XML. As you can't have 2 array items with the same keys (`<option name="mounts" .../><option name="mounts" .../>` is converted to `['mounts' => ..., 'mounts' => ...]`).

File: DependencyInjection/CmfResourceExtension.php (hunk 2 above)
Comment (WouterJ) [inline DependencyInjection/CmfResourceExtension.php]: Yeah, I was also thinking about using factories or the like for this.

**Comment by dantleech** [comment]
For me its good, if you can fix the tests feel free to merge.

**Comment by WouterJ** [comment]
I finally managed to have PHP 5.3 support (some crazy foreach problems...). Thanks for the review @dantleech !

**Pull request merged**